src/java.base/share/classes/jdk/internal/classfile/snippet-files/PackageSnippets.java

```diff
@@ -0,0 +1,308 @@
+import jdk.internal.classfile.CodeModel;
+import jdk.internal.classfile.CodeTransform;
+import jdk.internal.classfile.FieldModel;
+import jdk.internal.classfile.MethodElement;
+import jdk.internal.classfile.MethodModel;
+import jdk.internal.classfile.Opcode;
+import jdk.internal.classfile.TypeKind;
+import jdk.internal.classfile.instruction.FieldInstruction;
+import jdk.internal.classfile.instruction.InvokeInstruction;
+
+import static java.util.stream.Collectors.toSet;
+import jdk.internal.classfile.components.ClassRemapper;
+import jdk.internal.classfile.components.CodeLocalsShifter;
+import jdk.internal.classfile.components.CodeRelabeler;
+import jdk.internal.classfile.instruction.ReturnInstruction;
+import jdk.internal.classfile.instruction.StoreInstruction;
+
+class PackageSnippets {
+    void enumerateFieldsMethods1(byte[] bytes) {
+        // @start region="enumerateFieldsMethods1"
+        ClassModel cm = Classfile.parse(bytes);
+        for (FieldModel fm : cm.fields())
+            System.out.printf("Field %s%n", fm.fieldName().stringValue());
+        for (MethodModel mm : cm.methods())
+            System.out.printf("Method %s%n", mm.methodName().stringValue());
+        // @end
+    }
+
+    void enumerateFieldsMethods2(byte[] bytes) {
+        // @start region="enumerateFieldsMethods2"
+        ClassModel cm = Classfile.parse(bytes);
+        for (ClassElement ce : cm) {
+            switch (ce) {
+                case MethodModel mm -> System.out.printf("Method %s%n", mm.methodName().stringValue());
+                case FieldModel fm -> System.out.printf("Field %s%n", fm.fieldName().stringValue());
+                default -> { }
+            }
+        }
+        // @end
+    }
+
+    void gatherDependencies1(byte[] bytes) {
+        // @start region="gatherDependencies1"
+        ClassModel cm = Classfile.parse(bytes);
+        Set<ClassDesc> dependencies = new HashSet<>();
+
+        for (ClassElement ce : cm) {
+            if (ce instanceof MethodModel mm) {
+                for (MethodElement me : mm) {
+                    if (me instanceof CodeModel xm) {
+                        for (CodeElement e : xm) {
+                            switch (e) {
+                                case InvokeInstruction i -> dependencies.add(i.owner().asSymbol());
+                                case FieldInstruction i -> dependencies.add(i.owner().asSymbol());
+                                default -> { }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        // @end
+    }
+
+    void gatherDependencies2(byte[] bytes) {
+        // @start region="gatherDependencies2"
+        ClassModel cm = Classfile.parse(bytes);
+        Set<ClassDesc> dependencies = cm.elementStream()
+                                        .filter(ce -> ce instanceof MethodModel)
+                                        .flatMap(ce -> ((MethodModel) ce).elementStream())
```

Comment:
You could potentially collapse this into a single `flatMap` and avoid the cast:
```
.flatMap(ce -> ce instanceof MethodMethod mm ? mm.elementStream() : Stream.empty())
```

Reply:
fixed, thanks.